### PR TITLE
onnxruntime: 1.16.3 -> 1.18.1

### PR DESCRIPTION
**Pull request by jkachmar**
Release notes:
- https://github.com/microsoft/onnxruntime/releases/tag/v1.17.0
- https://github.com/microsoft/onnxruntime/releases/tag/v1.17.1
- https://github.com/microsoft/onnxruntime/releases/tag/v1.17.3
- https://github.com/microsoft/onnxruntime/releases/tag/v1.18.0
- https://github.com/microsoft/onnxruntime/releases/tag/v1.18.1

continuation of my last PR, which i somehow messed up so bad that GitHub automatically pinged everyone for some reason i cannot even remotely discern...?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by jkachmar** [comment]
@jonringer since you reviewed the last PR

**Comment by jkachmar** [comment]
`aarch64-linux` failures:
```
The following tests FAILED:
          4 - onnxruntime_shared_lib_test (Failed)
          6 - onnxruntime_customopregistration_test (Failed)
Errors while running CTest
make: *** [Makefile:71: test] Error 8
error: builder for '/nix/store/ngbnf85zfls5fxrjlyb1lc1s912zn2a4-onnxruntime-1.18.0.drv' failed with exit code 2;
       last 10 log lines:
       >
       > 71% tests passed, 2 tests failed out of 7
       >
       > Total Test time (real) =  68.51 sec
       >
       > The following tests FAILED:
       >       4 - onnxruntime_shared_lib_test (Failed)
       >       6 - onnxruntime_customopregistration_test (Failed)
       > Errors while running CTest
       > make: *** [Makefile:71: test] Error 8
```

**Comment by jkachmar** [comment]
looks like it was building previously so i guess the next step would be trying to figure out wtf the issue is: https://hydra.nixos.org/build/262732802

**Comment by mweinelt** [comment]
In wonder if we even expose the sysfs in the build sandbox.

Please rebase and unvendor cpuinfo. Worst case we disable tests on aarch64-linux.

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 319880` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages marked as broken and skipped:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>piper-train</li>
    <li>piper-train.dist</li>
    <li>python311Packages.tf2onnx</li>
    <li>python311Packages.tf2onnx.dist</li>
    <li>python312Packages.faster-whisper</li>
    <li>python312Packages.faster-whisper.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
    <li>wyoming-faster-whisper</li>
    <li>wyoming-faster-whisper.dist</li>
  </ul>
</details>
<details>
  <summary>13 packages failed to build:</summary>
  <ul>
    <li>opencomposite-helper</li>
    <li>python311Packages.chromadb</li>
    <li>python311Packages.chromadb.dist</li>
    <li>python311Packages.langchain-chroma</li>
    <li>python311Packages.langchain-chroma.dist</li>
    <li>python311Packages.mmcv</li>
    <li>python311Packages.mmcv.dist</li>
    <li>python312Packages.chromadb</li>
    <li>python312Packages.chromadb.dist</li>
    <li>python312Packages.langchain-chroma</li>
    <li>python312Packages.langchain-chroma.dist</li>
    <li>python312Packages.mmcv</li>
    <li>python312Packages.mmcv.dist</li>
  </ul>
</details>
<details>
  <summary>45 packages built:</summary>
  <ul>
    <li>aitrack</li>
    <li>deface</li>
    <li>deface.dist</li>
    <li>envision</li>
    <li>livecaptions</li>
    <li>local-ai</li>
    <li>maa-assistant-arknights</li>
    <li>maa-cli</li>
    <li>magika (python312Packages.magika)</li>
    <li>magika.dist (python312Packages.magika.dist)</li>
    <li>monado</li>
    <li>obs-studio-plugins.obs-backgroundremoval</li>
    <li>onnxruntime</li>
    <li>onnxruntime.dev</li>
    <li>onnxruntime.dist</li>
    <li>piper-phonemize</li>
    <li>piper-tts</li>
    <li>python311Packages.fastembed</li>
    <li>python311Packages.fastembed.dist</li>
    <li>python311Packages.faster-whisper</li>
    <li>python311Packages.faster-whisper.dist</li>
    <li>python311Packages.insightface</li>
    <li>python311Packages.insightface.dist</li>
    <li>python311Packages.magika</li>
    <li>python311Packages.magika.dist</li>
    <li>python311Packages.onnxruntime</li>
    <li>python311Packages.onnxruntime.dist</li>
    <li>python311Packages.piper-phonemize</li>
    <li>python311Packages.piper-phonemize.dist</li>
    <li>python311Packages.pysilero-vad</li>
    <li>python311Packages.pysilero-vad.dist</li>
    <li>python311Packages.rapidocr-onnxruntime</li>
    <li>python311Packages.rapidocr-onnxruntime.dist</li>
    <li>python312Packages.fastembed</li>
    <li>python312Packages.fastembed.dist</li>
    <li>python312Packages.insightface</li>
    <li>python312Packages.insightface.dist</li>
    <li>python312Packages.onnxruntime</li>
    <li>python312Packages.onnxruntime.dist</li>
    <li>python312Packages.piper-phonemize</li>
    <li>python312Packages.piper-phonemize.dist</li>
    <li>python312Packages.pysilero-vad</li>
    <li>python312Packages.pysilero-vad.dist</li>
    <li>python312Packages.rapidocr-onnxruntime</li>
    <li>python312Packages.rapidocr-onnxruntime.dist</li>
  </ul>
</details>

File: pkgs/development/libraries/onnxruntime/default.nix [hunk 1]
```diff
@@ -102,6 +107,13 @@ effectiveStdenv.mkDerivation rec {
     # - use MakeAvailable instead of the low-level Populate,
     # - use Eigen3::Eigen as the target name (as declared by libeigen/eigen).
     ./0001-eigen-allow-dependency-injection.patch
+    # Incorporate a patch that has landed upstream which exposes new
+    # 'abseil-cpp' libraries & modifies the 're2' CMakeLists to fix a
+    # configuration error that around missing 'gmock' exports.
+    #
+    # This can be dropped once 'onnxruntime' cuts a new release & we update.
+    # https://github.com/microsoft/onnxruntime/commit/b522df0ae477e59f60acbe6c92c8a64eda96cace
+    ./update-re2.patch
```
Comment (jkachmar) [inline pkgs/development/libraries/onnxruntime/default.nix:116]: @mweinelt should this comment be updated? i assumed that `onnxruntime` would incorporate changes like this one into minor releases, but `1.18.1` didn't have this fix so i guess it won't land until `1.19.x`?

**Comment by jkachmar** [comment]
are all the packages that fail to build legitimate failures as a result of this PR?

i tried to build `python311.pkgs.chromadb` & got this error:
```
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for chromadb-0.5.3-py3-none-any.whl
  - chroma-hnswlib==0.7.3 not satisfied by version 0.7.5
```

if i update `chromadb` to `0.5.4` then it passes the imports check phase & moves on to running tests.

**Comment by drupol** [comment]
> are all the packages that fail to build legitimate failures as a result of this PR?
> 
> i tried to build `python311.pkgs.chromadb` & got this error:
> ```
> Executing pythonRuntimeDepsCheck
> Checking runtime dependencies for chromadb-0.5.3-py3-none-any.whl
>   - chroma-hnswlib==0.7.3 not satisfied by version 0.7.5
> ```
> 
> if i update `chromadb` to `0.5.4` then it passes the imports check phase & moves on to running tests.
> 
> <details> <summary> update-chromadb-0.5.4.patch </summary>
> 
> ```
> diff --git a/pkgs/development/python-modules/chromadb/default.nix b/pkgs/development/python-modules/chromadb/default.nix
> index 50cae5fb0..e848beaa9 100644
> --- a/pkgs/development/python-modules/chromadb/default.nix
> +++ b/pkgs/development/python-modules/chromadb/default.nix
> @@ -50,7 +50,7 @@
> 
>  buildPythonPackage rec {
>    pname = "chromadb";
> -  version = "0.5.3";
> +  version = "0.5.4";
>    pyproject = true;
> 
>    disabled = pythonOlder "3.9";
> @@ -59,13 +59,13 @@ buildPythonPackage rec {
>      owner = "chroma-core";
>      repo = "chroma";
>      rev = "refs/tags/${version}";
> -    hash = "sha256-czDL2b+Jj7mrYZCTfnaZArkOHBaWyTV0BTE2wvykHps=";
> +    hash = "sha256-wzfzuWuNqLAjfAZC38p1iTtJHez/pJ9Ncgeo23o1dMo=";
>    };
> 
>    cargoDeps = rustPlatform.fetchCargoTarball {
>      inherit src;
>      name = "${pname}-${version}";
> -    hash = "sha256-eTVT1yowuDsajjceWojdUdX466FKneUt1i5QipBFdp4=";
> +    hash = "sha256-0OE2i29oE6RpJRswQWI8+5dbA6lOWd3nhqe1RGlnjhk=";
>    };
> 
>    pythonRelaxDeps = [ "orjson" ];
> ```
> 
> </details>

Fixed in https://github.com/NixOS/nixpkgs/pull/326337
Can you please review it so we can merge it ?

**Comment by mweinelt** [comment]
> are all the packages that fail to build legitimate failures as a result of this PR?

No, they were already broken on the base branch.

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 319880` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>piper-train</li>
    <li>piper-train.dist</li>
    <li>python311Packages.tf2onnx</li>
    <li>python311Packages.tf2onnx.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>
<details>
  <summary>9 packages failed to build:</summary>
  <ul>
    <li>opencomposite-helper</li>
    <li>python311Packages.chromadb</li>
    <li>python311Packages.chromadb.dist</li>
    <li>python311Packages.langchain-chroma</li>
    <li>python311Packages.langchain-chroma.dist</li>
    <li>python311Packages.mmcv</li>
    <li>python311Packages.mmcv.dist</li>
    <li>python312Packages.mmcv</li>
    <li>python312Packages.mmcv.dist</li>
  </ul>
</details>
<details>
  <summary>55 packages built:</summary>
  <ul>
    <li>aitrack</li>
    <li>deface</li>
    <li>deface.dist</li>
    <li>envision</li>
    <li>livecaptions</li>
    <li>local-ai</li>
    <li>maa-assistant-arknights</li>
    <li>maa-cli</li>
    <li>magika (python312Packages.magika)</li>
    <li>magika.dist (python312Packages.magika.dist)</li>
    <li>monado</li>
    <li>obs-studio-plugins.obs-backgroundremoval</li>
    <li>onnxruntime</li>
    <li>onnxruntime.dev</li>
    <li>onnxruntime.dist</li>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>piper-phonemize</li>
    <li>piper-tts</li>
    <li>python311Packages.fastembed</li>
    <li>python311Packages.fastembed.dist</li>
    <li>python311Packages.faster-whisper</li>
    <li>python311Packages.faster-whisper.dist</li>
    <li>python311Packages.insightface</li>
    <li>python311Packages.insightface.dist</li>
    <li>python311Packages.magika</li>
    <li>python311Packages.magika.dist</li>
    <li>python311Packages.onnxruntime</li>
    <li>python311Packages.onnxruntime.dist</li>
    <li>python311Packages.piper-phonemize</li>
    <li>python311Packages.piper-phonemize.dist</li>
    <li>python311Packages.pysilero-vad</li>
    <li>python311Packages.pysilero-vad.dist</li>
    <li>python311Packages.rapidocr-onnxruntime</li>
    <li>python311Packages.rapidocr-onnxruntime.dist</li>
    <li>python312Packages.chromadb</li>
    <li>python312Packages.chromadb.dist</li>
    <li>python312Packages.fastembed</li>
    <li>python312Packages.fastembed.dist</li>
    <li>python312Packages.faster-whisper</li>
    <li>python312Packages.faster-whisper.dist</li>
    <li>python312Packages.insightface</li>
    <li>python312Packages.insightface.dist</li>
    <li>python312Packages.langchain-chroma</li>
    <li>python312Packages.langchain-chroma.dist</li>
    <li>python312Packages.onnxruntime</li>
    <li>python312Packages.onnxruntime.dist</li>
    <li>python312Packages.piper-phonemize</li>
    <li>python312Packages.piper-phonemize.dist</li>
    <li>python312Packages.pysilero-vad</li>
    <li>python312Packages.pysilero-vad.dist</li>
    <li>python312Packages.rapidocr-onnxruntime</li>
    <li>python312Packages.rapidocr-onnxruntime.dist</li>
    <li>wyoming-faster-whisper</li>
    <li>wyoming-faster-whisper.dist</li>
  </ul>
</details>

**Pull request merged**